framework/wazuh/agent.py

```diff
@@ -396,7 +396,8 @@ def get_agents_in_group(group_list: list, offset: int = 0, limit: int = common.D
     if group_list[0] not in system_groups:
         raise WazuhResourceNotFound(1710)
 
-    q = 'group=' + group_list[0] + (';' + q if q else '')
+    q_group = 'group=' + group_list[0]
+    q = q_group + ';' + (q if q.startswith('(') and q.endswith(')') else f'({q})') if q else q_group
```

Comment:
I think it could be easier to read if we do

```suggestion
    if q and not q.startswith('(') and not q.endswith(')'):
        q = f'({q})'
    q = f'group={group_list[0]};' + q
```

Reply:
Simplified logic:
https://github.com/wazuh/wazuh/blob/ca8901ea4735ed4d252571ca378063bb780fdf21/framework/wazuh/agent.py#L399-L402